docs/how-to-use-custom-plugins/README.md

```diff
@@ -0,0 +1,193 @@
+# How to use your custom plugins in the simulator
+
+This doc describes how to use your custom plugins in the simulator.
+
+## 1. Add your custom plugin's registry in OutOfTreeRegistries function.
+
+Please add your custom plugin's registry in `OutOfTreeRegistries` function in config package here:
+
+[kube-scheduler-simulator/scheduler/config/plugin.go](../../config/plugin.go)
+
+
+## 2. Add your custom plugin in AdditionalFilterPlugins or AdditionalScorePlugins function.
+
+**only Filter and Score plugins**
+
+Please add your custom plugin in `AdditionalFilterPlugins` function or `AdditionalScorePlugins` function in config package here:
+
+[kube-scheduler-simulator/scheduler/config/plugin.go](../../config/plugin.go)
+
+## 3. Configure the scheduler to enable your custom plugin 
+
+You can configure the scheduler to use your custom plugins through KubeSchedulerConfiguration.
+
+[Scheduler Configuration | Kubernetes](https://kubernetes.io/docs/reference/scheduling/config/)
+
+You can change the scheduler configuration in Web UI or by passing a KubeSchedulerConfiguration file via the environment variable `KUBE_SCHEDULER_CONFIG_PATH`.
+
+## Example
+
+We will explain the case where you want to add [nodenumber](./nodenumber/plugin.go) plugin as example.
+
+The nodenumber plugin is a example plugin that favors nodes that have the number suffix which is the same as the number suffix of pod name.
+And we can configure it via `NodeNumberArgs`.
+
+First, we need to add registry for the nodenumber plugin to `OutOfTreeRegistries`.
+
+```go
+func OutOfTreeRegistries() runtime.Registry {
+	return runtime.Registry{
+		// Note: add your original score plugins registries here.
+		nodenumber.Name: nodenumber.New,
+	}
+}
+```
+
+Next, add nodenumber to `AdditionalScorePlugins` since nodenumber plugin is Score plugin.
+
+```go
+func AdditionalScorePlugins() []v1beta2.Plugin {
+	var weight1 int32 = 1
+	return []v1beta2.Plugin{
+		// Note: add your original score plugins here.
+		{
+			Name:   nodenumber.Name,
+			Weight: &weight1,
```

Comment:
What is the relationship to the weight value of KubeSchedulerConfiguration? Maybe it is good to add an explanation to the README.  (if not relevant, please ignore)